src/opentracing/instrument/ModuleHack.ts

```diff
@@ -0,0 +1,288 @@
+                }
+            } else if (node.type === self.syntax.CatchClause) {
+                let origCatch = node.body.source();
+                origCatch = origCatch.slice(1, origCatch.length - 1);
+                node.body.update('{\n' + ON_CATCH + '\n' + origCatch + '\n}');
+            }
+        });
+
+        return output.toString();
+    }
+
+    isFunctionNode(node: any) {
+        return (node.type === this.syntax.FunctionDeclaration ||
+            node.type === this.syntax.FunctionExpression || node.type === this.syntax.ArrowFunctionExpression) && node.range;
+    }
+
+    getFunctionName(node: any) {
+        if (!this.isFunctionNode(node)) {
+            return;
+        }
+        if (node.id) {
+            return node.id.name;
+        }
+
+        if (node.type === this.syntax.FunctionDeclaration) {
+            return '';
+        }
+
+        const parent = node.parent;
+        switch (parent.type) {
+            case this.syntax.AssignmentExpression:
+                if (parent.left.range) {
+                    return parent.left.source().replace(/"/g, '\\"');
+                }
+                break;
+
+            case this.syntax.VariableDeclarator:
+                return parent.id.name;
+
+            case this.syntax.CallExpression:
+                return parent.callee.id ? parent.callee.id.name : '[Anonymous]';
+            default:
+                if (typeof parent.length === 'number') {
+                    return parent.id ? parent.id.name : '[Anonymous]';
+                } else if (parent.key && parent.key.type === 'Identifier' &&
+                    parent.value === node && parent.key.name) {
+                    return parent.key.name;
+                }
+        }
+        return '[Anonymous]';
+    }
+
+    setGlobalFunction() {
+        const self: any = this;
+        global.__njsTraceEntry__ = function (args: any) {
+            try {
+                const span = self.tracer.startSpan(args.name);
+                const spanArguments: Argument[] = [];
+                if (args.args) {
+                    for (let i = 0; i < args.args.length; i++) {
+                        spanArguments.push(new Argument(args.argNames[i], typeof args.args[i], args.args[i]));
+                    }
+                }
+
+                span.setTag(ARGS_TAG_NAME, spanArguments);
+
+                return { name: args.name, file: args.file, fnLine: args.line, ts: Date.now()};
+            } catch (ex) {
+                console.log(ex);
+            }
+        };
+
+        global.__njsTraceExit__ = function (args: any) {
+            try {
+                const span = self.tracer.getActiveSpan();
+                console.log(args.exceptionValue);
+                if (!args.exception) {
+                    span.setTag(RETURN_VALUE_TAG_NAME, new ReturnValue(typeof args.returnValue, args.returnValue));
+                } else {
+                    span.setTag('error', args.exception);
```

Comment:
According to spec, this is the `error` tag definition:
> true if and only if the application considers the operation represented by the Span to have failed

So in case of exception, `error` must be `true`, and there should be error message with `error.message`